### PR TITLE
Add redrive policy to queue

### PR DESCRIPTION
**Pull request by MancunianSam**
The dead letter queue and retry policy within lambda doesn't work when
the event source in an sqs queue.

The relevant bit in the documentation is here:
https://docs.aws.amazon.com/lambda/latest/dg/invocation-retries.html

Event source mappings – Event source mappings that read from streams retry
the entire batch of items. Repeated errors block processing of the
affected shard until the error is resolved or the items expire.
To detect stalled shards, you can monitor the Iterator Age metric.

For event source mappings that read from a queue, you determine the length
of time between retries and destination for failed events by configuring
the visibility timeout and redrive policy on the source queue.

### CODE REVIEW

File: sqs/variables.tf [hunk 1]
```diff
@@ -29,4 +29,14 @@ variable "apply_resource" {
 variable "sqs_policy" {
   description = "allows a custom SQS policy to be set"
   default     = "default"
+}
+
+variable "dead_letter_queues" {
+  description = "The dead letter queue for failed messages to be sent to"
```
Comment (TomJKing) [inline sqs/variables.tf:35]: Should the description read "... queue**s** for failed ..." as it is an array

File: lambda/variables.tf [hunk 2]
```diff
@@ -53,5 +53,5 @@ variable "api_url" {
 
 variable "keycloak_backend_checks_client_secret" {
   description = "Keycloak backend checks client secret"
-  default = ""
+  default     = ""
 }
```
Comment (TomJKing) [inline lambda/variables.tf:57]: Add line break at end of file

File: sqs/variables.tf [hunk 3]
```diff
@@ -29,4 +29,14 @@ variable "apply_resource" {
...
+  default     = []
+}
+
+variable "redrive_maximum_receives" {
+  description = "The maximum number of receives if using a redrive policy"
+  default     = 0
 }
```
Comment (TomJKing) [inline sqs/variables.tf:42]: Add line break at end of file

File: sqs/variables.tf (hunk 1 above)
Comment (MancunianSam) [reply sqs/variables.tf:35]: It shouldn't be an array. I was originally trying to use a dynamic block which takes an array but that didn't work. You can only have one dlq for an sqs queue so this makes more sense as a string. I've changed the variable and changed the name.

File: lambda/variables.tf (hunk 2 above)
Comment (MancunianSam) [reply lambda/variables.tf]: I've added it in

File: sqs/variables.tf (hunk 3 above)
Comment (MancunianSam) [reply sqs/variables.tf]: I've added another line break

**Pull request merged**